CMakeLists.txt

```diff
@@ -590,11 +582,14 @@ if (CUDAQ_ENABLE_PYTHON)
     message(FATAL_ERROR "Applying patch to submodule failed with ${GIT_PATCH_RESULT}, please update patch")
   endif()
   
+  # Regarding the use of PyBind, we need to be careful that the same STL is used for any 
+  # Python bindings generated as part of the CUDA-Q build and bindings generated for 
+  # third party CUDA-Q libraries; see also https://github.com/pybind/pybind11/issues/1262
```

Comment:
It looks like that upstream issue was marked as resolved. Do we need to update which version we are using?

Reply:
Same here, and the pybind update may potentially require a bit of additional work (last time I had tried to set it to the latest version that ran into issues)